### PR TITLE
overlord/snapstate: ensure calls to store are done without the state lock held

### PR DESCRIPTION
**Pull request by chipaca**

### CODE REVIEW

File: overlord/snapstate/backend_test.go [hunk 1]
```diff
@@ -54,9 +54,13 @@ type fakeStore struct {
 	fakeBackend         *fakeSnappyBackend
 	fakeCurrentProgress int
 	fakeTotalProgress   int
+	state               *state.State
 }
 
 func (f *fakeStore) Snap(name, channel string, devmode bool, user *auth.UserState) (*snap.Info, error) {
+	f.state.Lock()
+	defer f.state.Unlock()
```
Comment (pedronis) [inline overlord/snapstate/backend_test.go]: any reason not to just do f.state.Unlock() directly, no defer?

**Comment by pedronis** [comment]
looks good, one small wondering (that applies to a couple of places)

**Comment by pedronis** [comment]
+1

**Comment by matiasb** [comment]
:+1: LGTM

**Pull request merged**